merkle/hashers/tree_hasher.go

```diff
@@ -42,3 +48,47 @@ type MapHasher interface {
 	// TODO(gbelvin): Replace Size() with BitLength().
 	Size() int
 }
+
+var (
+	maxHashers = trillian.HashStrategy(len(trillian.HashStrategy_value))
```

Comment:
I'd suggest a map[HashStrategy]XHasher instead. No need to worry about proto tags that way.